proteuslib/flowsheets/full_treatment_train/example_models/nanofiltration_ZO.py

```diff
@@ -0,0 +1,409 @@
+
+    def initialize(
+            blk,
+            state_args=None,
+            outlvl=idaeslog.NOTSET,
+            solver=None,
+            optarg=None):
+        """
+        General wrapper for pressure changer initialization routines
+
+        Keyword Arguments:
+            state_args : a dict of arguments to be passed to the property
+                         package(s) to provide an initial state for
+                         initialization (see documentation of the specific
+                         property package) (default = {}).
+            outlvl : sets output level of initialization routine
+            optarg : solver options dictionary object (default=None)
+            solver : str indicating which solver to use during
+                     initialization (default = None)
+
+        Returns: None
+        """
+        init_log = idaeslog.getInitLogger(blk.name, outlvl, tag="unit")
+        solve_log = idaeslog.getSolveLogger(blk.name, outlvl, tag="unit")
+        # Set solver options
+        opt = get_solver(solver, optarg)
+
+        # ---------------------------------------------------------------------
+        # Initialize holdup block
+        flags = blk.feed_side.initialize(
+            outlvl=outlvl,
+            optarg=optarg,
+            solver=solver,
+            state_args=state_args,
+        )
+        init_log.info_high("Initialization Step 1 Complete.")
+        # ---------------------------------------------------------------------
+        # Initialize permeate
+        # Set state_args from inlet state
+        if state_args is None:
+            state_args = {}
+            state_dict = blk.feed_side.properties_in[
+                blk.flowsheet().config.time.first()].define_port_members()
+
+            for k in state_dict.keys():
+                if state_dict[k].is_indexed():
+                    state_args[k] = {}
+                    for m in state_dict[k].keys():
+                        state_args[k][m] = state_dict[k][m].value
+                else:
+                    state_args[k] = state_dict[k].value
+
+        blk.properties_permeate.initialize(
+            outlvl=outlvl,
+            optarg=optarg,
+            solver=solver,
+            state_args=state_args,
+        )
+        init_log.info_high("Initialization Step 2 Complete.")
+
+        # ---------------------------------------------------------------------
+        # Solve unit
+        with idaeslog.solver_log(solve_log, idaeslog.DEBUG) as slc:
+            res = opt.solve(blk, tee=slc.tee)
+        init_log.info_high(
+            "Initialization Step 3 {}.".format(idaeslog.condition(res)))
+
+        # ---------------------------------------------------------------------
+        # Release Inlet state
+        blk.feed_side.release_state(flags, outlvl + 1)
+        init_log.info(
+            "Initialization Complete: {}".format(idaeslog.condition(res))
+        )
+
+    def _get_performance_contents(self, time_point=0):
```

Comment:
It would be good to fill in `_get_performance_contents` and also add `_get_stream_table_contents` to use the `report()` method with this NF model. Since we'd probably want to this model merged into the Sept release with documentation, reporting capability should also be enabled, and it isn't very difficult to do so. @TimBartholomew - I can add reporting functionality here, unless you have objections to doing so.

Reply:
@adam-a-a please add it.